src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/IdGeneratorService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.companieshouse.api.strikeoffobjections.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IIdGeneratorService;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+@Service
+public class IdGeneratorService implements IIdGeneratorService {
+
+    private static final String ID_PREFIX = "OBJ-";
+
+    private final Supplier<LocalDateTime> dateTimeSupplier;
+
+    @Autowired
+    public IdGeneratorService(Supplier<LocalDateTime> dateTimeSupplier) {
+        this.dateTimeSupplier = dateTimeSupplier;
+    }
+
+    @Override
+    public String generateId() {
+        LocalDate dateNow = dateTimeSupplier.get().toLocalDate();
+        String now = Long.toHexString(dateNow.toEpochDay());
+        String uuid = now + UUID.randomUUID().toString();
```

Comment:
Var name here isn't strictly correct - it's 'now' + a UUID. Could we rename? Maybe just id would do?